### PR TITLE
require wave_gazebo_plugins to use c++17

### PR DESCRIPTION
**Pull request by acxz**
Solves #471

### CODE REVIEW

**Comment by acxz** [comment]
@M1chaelM just a friendly ping

**Comment by M1chaelM** [comment]
@acxz We are a bit occupied with other priorities at the moment, so I don't have a timeline for when we will get to review your feature requests. Eventually, I hope we'll have the chance to look through them and add them to a milestone if appropriate. If you need them in the near term, I'd recommend working from your fork.

**Comment by M1chaelM** [comment]
> Would you still be okay with me opening more issues and PRs?

Yes, that's fine. If you have a lot of stuff though it might be good for us to get a sense of what you're working on and what your requirements are so we can plan for support. If we understand the high level objectives it will help us figure out how to prioritize.

**Comment by acxz** [comment]
> If we understand the high level objectives it will help us figure out how to prioritize.

Sure. Essentially, I'm trying to solve #465 . I have a custom vehicle that I'd like to use in VRX. Previously I was using UUV, however, UUV does not have wave or wind simulation. As its name suggests its primary focus is for underwater vehicles (not the mention that it is essentially in archive mode, since the maintainers have not able to successfully been able to hand off leadership to a someone else). VRX is the only currently maintained simulation environment for USVs in Gazebo classic.

However, VRX itself is essentially hardcoded to work for two hull half cylinder type vehicles. The tugboat in #476, the custom vehicle in #434, and #465 all illustrate the community's need to generalize VRX to various USVs.

TLDR; add generic USV simulation to the Gazebo Classic ecosystem

**Comment by M1chaelM** [comment]
> However, VRX itself is essentially hardcoded to work for two hull half cylinder type vehicles. The tugboat in #476, the custom vehicle in #434, and #465 all illustrate the community's need to generalize VRX to various USVs.
> 
> TLDR; add generic USV simulation to the Gazebo Classic ecosystem

Thanks! This was very helpful. I agree it would be useful to make adding new vehicles easier. Do you need to use Gazebo Classic, or would Gazebo Sim work?

Can you point me to the custom vehicle that you want to use? I'd like to aggregate a few examples.

**Comment by acxz** [comment]
> Do you need to use Gazebo Classic, or would Gazebo Sim work?

To be honest, I would like to use Gazebo Sim. However, currently Gazebo Sim does not have released plugins as high fidelity as Gazebo Classic for hydrodynamics work. One is in development by srmainwaring, which I am patiently waiting for. To be fair, VRX also has some fidelity that is lacking for my use cases, but I as well as others are also addressing this gap in several open PRs.

> custom vehicle that you want to use

Clearpath's Heron: https://clearpathrobotics.com/heron-contact-us/
URDF files: https://github.com/heron/heron/tree/melodic-devel/heron_description

**Comment by caguero** [comment]
@acxz , now that the main functionality has been ported to Gazebo and ROS 2 in `main`. Are you interested in retargeting your open PRs to main? We'll start reviewing them now that we have a stable codebase after the `2.0` release.

Thanks!

**Comment by caguero** [comment]
@acxz , friendly ping. What do you think about retargeting your PRs?

**Comment by acxz** [comment]
@caguero thanks for the ping, my project is still on ros-noetic and we haven't moved over to ROS 2 yet. Once we do or if I get some extra time I'll start redoing my PRs.

**Comment by caguero** [comment]
@acxz , we got some time to help with your contributions. Are you willing to retarget them to the `jazzy` branch?

**Comment by caguero** [comment]
> @acxz , we got some time to help with your contributions. Are you willing to retarget them to the `jazzy` branch?

@acxz , friendly ping. We can work with you on your PRs. Do you have some time to spend with us on them?

**Comment by caguero** [comment]
Closing in favor of #828 .
